### PR TITLE
Update docs for query performance mode

### PR DESCRIPTION
**Pull request by CamronStaley**
## What changes are proposed in this pull request?

Update docs for query performance mode

## How is this patch tested? If it is not, please explain why.

local dev

### Is this a user-facing change that should be mentioned in the release notes?

-   [x] No. You can skip the rest of this section.
-   [ ] Yes. Give a description of this change to be included in the release
        notes for FiftyOne users.

(Details in 1-2 sentences. You can just refer to another PR with a description
if this PR is part of a larger change.)

### What areas of FiftyOne does this PR affect?

-   [ ] App: FiftyOne application changes
-   [ ] Build: Build and test infrastructure changes
-   [ ] Core: Core `fiftyone` Python library changes
-   [x] Documentation: FiftyOne documentation changes
-   [ ] Other

## Release Notes

- **New Features**
  - Introduced a `Query Performance` mode for optimizing query handling in the app.
  - Added configuration options for `default_query_performance` and `enable_query_performance`.

- **Bug Fixes**
  - Enhanced logic for displaying the `QueryPerformanceToast` to show only when relevant data is available.

- **Documentation**
  - Updated user guides to reflect changes in session management and performance optimization.
  - Improved clarity and organization of configuration documentation.

### CODE REVIEW

File: app/packages/core/src/components/QueryPerformanceToast.tsx [hunk 1]
```diff
@@ -57,7 +58,7 @@ const QueryPerformanceToast = ({
     throw new Error("no query performance element");
   }
 
-  if (!shown || disabled) {
+  if (!shown || disabled || !queryPerformance) {
```
Comment (minhtuev) [inline app/packages/core/src/components/QueryPerformanceToast.tsx:61]: if QP mode is False, we won't be showing the toast.

File: docs/source/user_guide/app.rst [hunk 2]
```diff
@@ -519,32 +519,26 @@ You can conveniently reset the sidebar groups to their default state by setting
...
-enabled either by adding a `lightning_threshold` to your
+Query Performance mode is a performant sidebar setting for larger datasets that can be
+enabled either by adding a `default_query_performance` to your
 :ref:`App config <configuring-fiftyone-app>`, or for a particular dataset by
 clicking on the "Gear" icon above the sample grid in the App.
 
 .. note::
 
-    When lightning mode is enabled through the "Gear" icon in the App, the
+    When query performance mode is enabled through the "Gear" icon in the App, the
```
Comment (minhtuev) [inline docs/source/user_guide/app.rst:534]: Do we want to keep Query Performance or query performance for consistency?

**Review by minhtuev — APPROVED** [review]
Small comments otherwise LGTM

**Review by findtopher — APPROVED** [review]
Looks like a reasonable change post dev-cut to me

File: docs/source/user_guide/config.rst [hunk 3]
```diff
@@ -804,7 +809,9 @@ You can print your App config at any time via the Python library and the CLI:
             ],
             "colorscale": "viridis",
             "grid_zoom": 5,
-            "lightning_threshold": null,
+            "default_query_performance": true,
+            "disable_frame_filtering": false,
+            "enable_query_performance": true,
```
[the remark this answers is not in the document]
Comment (CamronStaley) [reply docs/source/user_guide/config.rst:814]: good bot

**Review by benjaminpkane — APPROVED** [review]

File: docs/source/user_guide/app.rst (hunk 2 above)
Comment (benjaminpkane) [reply docs/source/user_guide/app.rst:534]: Good point. Note the language can and may change after product feedback

**Review by findtopher — APPROVED** [review]
🔩

**Pull request merged**